py/server/deephaven/stream/kafka/producer.py

```diff
@@ -70,22 +75,28 @@ def produce(
             raise ValueError(
                 "at least one argument for 'key_spec' or 'value_spec' must be different from KeyValueSpec.IGNORE"
             )
-
-        kafka_config = j_properties(kafka_config)
-        runnable = _JKafkaTools.produceFromTable(
-            table.j_table,
-            kafka_config,
-            topic,
-            key_spec.j_object,
-            value_spec.j_object,
-            last_by_key_columns,
+        options = (
+            _JKafkaPublishOptions.builder()
+            .table(table.j_table)
+            .topic(topic)
+            .config(j_properties(kafka_config))
+            .keySpec(key_spec.j_object)
+            .valueSpec(value_spec.j_object)
+            .lastBy(last_by_key_columns and key_spec is not KeyValueSpec.IGNORE)
+            .publishInitial(publish_initial)
+            .build()
         )
 
+        with auto_locking_ctx(table):
+            runnable = _JKafkaTools.produceFromTable(options)
+
         def cleanup():
             try:
                 runnable.run()
             except Exception as ex:
-                raise DHError(ex, "failed to stop publishing to Kafka and the clean-up.") from ex
+                raise DHError(
+                    ex, "failed to stop publishing to Kafka and the clean-up."
+                ) from ex
```

Comment:
not sure why it needs to be multiple lines.

Reply:
I used black formatter; but until we apply it everywhere, I'll revert this part and make a single line.